### PR TITLE
Migrate System.Drawing.Common from dotnet/runtime into winforms

### PR DESCRIPTION
**Pull request by ViktorHofer**
An announcement for the code move will be following soon.

**Important: All commits except the last two are migrated from dotnet/runtime and shouldn't be reviewed.**

Commits are migrated on 2/13/2023. Here's the script that I created to migrate them: https://gist.github.com/ViktorHofer/2be5314439e75860326a60cee6523bef

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8633)

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
FYI: This is an engineering change to align the library with the primary `System.Drawing` maintainers (the WinForms team). Only `System.Drawing.Common` is moving. `System.Drawing.Primitives` (`Color`, `Rectangle`, etc.) will remain in the .NET runtime repo.

File: src/Common/src/NullableAttributes.cs [hunk 1]
```diff
@@ -0,0 +1,196 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Diagnostics.CodeAnalysis
```
Comment (gpetrou) [inline src/Common/src/NullableAttributes.cs:4]: Why do we need to add these files?

**Comment by JeremyKuhne** [comment]
@ViktorHofer - this is looking good! The changes so far make sense to me.

**Comment by elachlan** [comment]
Are we blocked by https://github.com/microsoft/win32metadata/issues/1287?

**Comment by ViktorHofer** [comment]
I verified that the produced package contains the correct assemblies, placeholder files, msbuild targets files and intellisense documentation files. I'm verifying right now if the produced Microsoft.Private.Winforms transport package includes System.Drawing.Common assets.

**Comment by ViktorHofer** [comment]
The PR is now ready to be reviewed and merged. @JeremyKuhne can you please take a look?

Summary of the follow-up work post-merge:
1. Make sure that WindowsDesktop.Ref picks up the right System.Drawing.Common assemblies.
2. Verify that packages flow correctly and that we have access to the new System.Drawing.Common nuget package in the feeds.
3. Send a follow-up PR with the changes that went into S.D.C in runtime meanwhile.
4. Remove System.Drawing.Common from runtime and use the new package instead.
5. Migrate all the issues over to winforms
6. Publish the announcement issues

**Comment by ViktorHofer** [comment]
We don't want to squash the commits. Can someone create a merge commit please?

**Comment by dreddy-work** [comment]
> We don't want to squash the commits. Can someone create a merge commit please?

Merged without squash.

**Comment by dreddy-work** [comment]
@ViktorHofer, we are dealing with flaky tests in this repo but i will follow-up CI flows into WPF->Desktop.

**Pull request merged**